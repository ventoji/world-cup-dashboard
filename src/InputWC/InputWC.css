.input-wc {
  margin-bottom: '16px';
}

.input-wc__error {
  color: red;
}

.input-wc__borders {
  border: 'solid 1px red';
}
