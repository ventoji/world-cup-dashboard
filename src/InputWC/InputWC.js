import React from 'react';
import LabelWC from '../LabelWC';
import PropTypes from 'prop-types';

/** InputWC field component */
const InputWC = ({
  htmlId,
  name,
  label,
  type = 'text',
  required = false,
  onChange,
  placeholder,
  value,
  error,
  children,
  ...props
}) => {
  return (
    <div style={{ marginBottom: '15px' }}>
      <LabelWC htmlFor={htmlId} label={label} required={required} />
      <input
        id={htmlId}
        type={type}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        style={error ? { border: 'solid 1px red' } : {}}
        {...props}
      />
      {children}
      {error && (
        <div className="input-wc__error" style={{ color: 'red' }}>
          {error}
        </div>
      )}
    </div>
  );
};

InputWC.propTypes = {
  /** Unique HTML ID. Used for tying label to HTML input. Handy hook for automated testing. */
  htmlId: PropTypes.string.isRequired,

  /** Input name. Recommend setting this to match object's property so a single change handler can be used. */
  name: PropTypes.string.isRequired,

  /** Input label */
  label: PropTypes.string.isRequired,

  /** Input type */
  type: PropTypes.oneOf(['text', 'number', 'password']),

  /** Mark label with asterisk if set to true */
  required: PropTypes.bool,

  /** Function to call onChange */
  onChange: PropTypes.func.isRequired,

  /** Placeholder to display when empty */
  placeholder: PropTypes.string,

  /** Value */
  value: PropTypes.any,

  /** String to display when error occurs */
  error: PropTypes.string,

  /** Child component to display next to the input */
  children: PropTypes.node,
};

InputWC.defaultProps = {
  htmlId: '1shdy4',
  type: 'text',
  name: '',
  label: 'home team',
  placeholder: 'value here',
  required: true,
  onChange: () => {},
};

export default InputWC;
