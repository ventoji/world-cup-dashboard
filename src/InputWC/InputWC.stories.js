import React from 'react';
import InputWC from './InputWC';
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs';

export default {
  title: 'WC Components/InpuWC',
  component: InputWC,
  decorators: [withKnobs],
};

const Template = (args) => <InputWC {...args} />;

export const Default = Template.bind({});

Default.args = {
  htmlFor: 'label',
  label: text('label', 'home Team'),
  name: text('label', 'hometeam'),
  onChange: action('cick'),
  type: 'text',
};

/** Numeric input a */
export const NumericInput = Template.bind({});
NumericInput.args = {
  htmlFor: 'labelNumeric',
  label: text('label', 'Score Team'),
  name: text('label', 'scoreteam'),
  onChange: action('cick'),
  type: 'number',
};
