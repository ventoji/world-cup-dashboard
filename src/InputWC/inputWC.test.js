import React from 'react';
import InputWC from './InputWC';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Input  Component', () => {
  let inputWC;

  beforeEach(() => {
    inputWC = shallow(
      <InputWC
        htmlId="labelElement"
        label="Home Team"
        name="hometeam"
        onChange={() => {}}
      />,
    );
  });

  it('Should be defined', () => {
    expect(InputWC).toBeDefined();
  });

  it('Should render properly', () => {
    let element = inputWC.find('input');

    expect(toJson(inputWC)).toMatchSnapshot();
    // input exists
    expect(element).toHaveLength(1);
    // input has attribute id equal to htmlId
    expect(element.props().id).toMatch(/labelElement/);
    expect(element.props().name).toMatch(/hometeam/);

    element = inputWC.find('Label');
    expect(element).toHaveLength(1);
  });

  it('should call mock function when input changes', () => {
    const mockHandleClick = jest.fn();

    inputWC = shallow(
      <InputWC
        htmlId="labelElement"
        label="Home Team"
        name="hometeam"
        onChange={mockHandleClick}
      />,
    );

    inputWC.find('input').simulate('change');

    //  inputWC.simulate('change');

    expect(mockHandleClick.mock.calls.length).toEqual(1);
    expect(mockHandleClick).toHaveBeenCalled();
  });
});
