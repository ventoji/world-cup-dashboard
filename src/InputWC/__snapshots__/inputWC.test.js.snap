// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input  Component Should render properly 1`] = `
<div
  style={
    Object {
      "marginBottom": "15px",
    }
  }
>
  <Label
    htmlFor="labelElement"
    label="Home Team"
    required={true}
  />
  <input
    id="labelElement"
    name="hometeam"
    onChange={[Function]}
    placeholder="value here"
    style={Object {}}
    type="text"
  />
</div>
`;
